plugin-server/src/main/ingestion-queues/session-recording/session-recordings-consumer.ts

```diff
@@ -304,7 +308,10 @@ export class SessionRecordingIngester {
             )
         }
 
-        await this.sessions[key]?.add(event)
+        await Promise.allSettled([
+            this.sessions[key]?.add(event),
+            this.overflowDetection?.observe(overflowKey, event.metadata.rawSize, event.metadata.timestamp),
```

Comment:
you could void this promise instead of waiting on it since it's ok for some to fail maybe, but it should be fast enough that that's bike-shedding and we'll see the impact if it isn't fast enough so ignore as nit-picking 🤣

Reply:
The only async part is the redis `ZADD` when triggering on the first time, the detection itself does not yield, so I don't think voiding it would help throughput 🤔 
I'd say let's rollout and re-evaluate if there's impact. When merging, overflow will be disabled, it'll be re-enabled when rolling out the new `CAPTURE_CONFIG_REDIS_HOST`